packages/flutter_tools/test/general.shard/build_system/targets/asset_transformer_test.dart

```diff
@@ -0,0 +1,377 @@
+
+    final FakeProcessManager processManager =
+        FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          artifacts.getArtifactPath(Artifact.engineDartBinary),
+          'run',
+          'my_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+          '-f',
+          '--my_option',
+          'my_option_value',
+        ],
+        onRun: (List<String> args) {
+          final ArgResults parsedArgs = (ArgParser()
+            ..addOption('input')
+            ..addOption('output')
+            ..addFlag('foo', abbr: 'f')
+            ..addOption('my_option'))
+            .parse(args);
+          fileSystem.file(parsedArgs['output']).createSync(recursive: true);
+        },
+      ),
+    ]);
+
+    final AssetTransformer transformer = AssetTransformer(
+      processManager: processManager,
+      logger: logger,
+      fileSystem: fileSystem,
+      dartBinaryPath: artifacts.getArtifactPath(Artifact.engineDartBinary),
+    );
+
+    await transformer.transformAsset(
+      asset: asset,
+      outputPath: outputPath,
+      workingDirectory: fileSystem.currentDirectory.path,
+      transformerEntries: <AssetTransformerEntry>[
+        const AssetTransformerEntry(
+          package: 'my_transformer',
+          args: <String>[
+            '-f',
+            '--my_option',
+            'my_option_value',
+          ],
+        )
+      ],
+    );
+
+    expect(processManager, hasNoRemainingExpectations);
+  });
+
+  testWithoutContext('logs useful error information when transformation process returns a nonzero exit code', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')..createSync();
+    const String outputPath = 'output.txt';
+
+    final String dartBinaryPath = artifacts.getArtifactPath(Artifact.engineDartBinary);
+    final FakeProcessManager processManager =
+        FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          dartBinaryPath,
+          'run',
+          'my_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+        ],
+        onRun: (List<String> args) {
+          final ArgResults parsedArgs = (ArgParser()
+            ..addOption('input')
+            ..addOption('output'))
+            .parse(args);
+          fileSystem.file(parsedArgs['output']).createSync(recursive: true);
+        },
+        exitCode: 1,
+        stdout: 'Beginning transformation',
+        stderr: 'Something went wrong',
+      ),
+    ]);
+
+    final AssetTransformer transformer = AssetTransformer(
+      processManager: processManager,
+      logger: logger,
+      fileSystem: fileSystem,
+      dartBinaryPath: dartBinaryPath,
+    );
+
+    await transformer.transformAsset(
```

Comment:
let's expect on return value

Reply:
Addressed in [nit pack 1](https://github.com/flutter/flutter/pull/143815/commits/898a4b6396aee4098bce759b35caec356c96809a)